### PR TITLE
Master: Use user-provided style rather than picking first

### PR DESCRIPTION
**Pull request by rdengate**
Styles can be passed as a parameter to style the WMS layer, but it is ignored for GetLegendGraphic, which currently uses the URL from the first style if a list is provided. Now it uses the specified style if provided, but otherwise chooses the first style as it previously did.

GEOGLAM requires this change in both master and newui. I'm intending to submit a separate PR for merge into newui.

### CODE REVIEW

File: lib/Models/WebMapServiceCatalogItem.js [hunk 1]
```diff
@@ -409,8 +409,24 @@ WebMapServiceCatalogItem.prototype.updateFromCapabilities = function(capabilitie
     }
 
     var legendUri, legendMimeType;
-    // There can be multiple styles - just get the first if so.
+    // If style is defined in parameters, use that, but only if a style with that name can be found.
```
Comment (kring) [inline lib/Models/WebMapServiceCatalogItem.js:412]: We don't verify that a style exists before passing it to `GetMap`, so I'm not sure why we would here.  IMO `parameters` should be passed through to the server without any judgement or interpretation, just do what the user tells us to do.  I think we should pass all of the `parameters` to `GetLegendGraphic` too, not just `style`.

Comment (rdengate) [inline lib/Models/WebMapServiceCatalogItem.js:412]: I didn't realize that was an option. It does sound like a much better idea. Thanks!

File: wwwroot/test/WMS/multiple_style_legend_url.xml [hunk 2]
```diff
@@ -102,7 +102,7 @@
                         <Format>image/gif</Format>
                         <OnlineResource xmlns:xlink="http://www.w3.org/1999/xlink"
                                         xlink:type="simple"
-                                        xlink:href="http://www.example.com/legendUrl"/>
+                                        xlink:href="http://www.example.com/request=GetLegendGraphic?firstUrl"/>
```
Comment (kring) [inline wwwroot/test/WMS/multiple_style_legend_url.xml]: This isn't actually a valid `GetLegendGraphic` URL.  It should be:
http://www.example.com?request=GetLegendGraphic&firstUrl

Comment (kring) [inline wwwroot/test/WMS/multiple_style_legend_url.xml]: Our regex currently matches anything with the string `GetLegendGraphic`, but we might tighten it up in the future, in which case it'd be nice if this test still passed.

**Comment by kring** [comment]
I made those small changes and merged it so we can include this in the NM release today.

**Pull request merged**